src/django/api/views/boundary.py

```diff
@@ -0,0 +1,69 @@
+from django.http import Http404
+from django.db.models import Prefetch
+
+from rest_framework.response import Response
+from rest_framework.exceptions import APIException
+from rest_framework.views import APIView
+from rest_framework.permissions import IsAuthenticated
+
+from ..models.user import Roles
+from ..models.boundary import Boundary
+from ..models.utility import Utility
+from ..models.submission import Submission
+
+from ..serializers.boundary import BoundaryDetailSerializer, BoundaryListSerializer
+
+
+def get_boundary_queryset_for_user(user):
+    if user.role == Roles.CONTRIBUTOR:
+        return Boundary.objects.filter(utility__in=user.utilities.all())
+
+    if user.role == Roles.VALIDATOR:
+        return Boundary.objects.filter(utility__state__in=user.states.all())
+
+    if user.role == Roles.ADMINISTRATOR:
+        return Boundary.objects.all()
+
+    raise RuntimeError('Invalid role: {}'.format(user.role))
+
+
+class BoundaryListView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, format=None):
+        boundaries = get_boundary_queryset_for_user(request.user)
+        boundaries = boundaries.select_related('utility')
+
+        requested_utilities = request.GET.get('utilities')
+        if requested_utilities:
+            for utility_id in requested_utilities:
+                if not request.user.has_access_to_utility(
+                    Utility.objects.get(pk=utility_id)
+                ):
+                    raise APIException('Cannot access this utility', 401)
```

Comment:
This could merit updating the ADR, but if we just raise `PermissionDenied`, DRF will convert this into a 403, which I think is more accurate than 401 given that this is an insufficient permission issue rather than incorrect creds issue.

Sample response from what I was just working on in 97ef19bcb2bea89287d131b990c39931f88ad72a:
```
GET /api/boundaries/1/reference-images/
HTTP 403 Forbidden
Allow: GET, POST, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "detail": "You do not have permission to perform this action."
}
```

Reply:
I'm okay with that. Looping in @rajadain